transport/transport.go

```diff
@@ -306,3 +320,58 @@ func NewMultiTransport(backends []http.RoundTripper,
 		Backends:        backends,
 		HandleResponses: responsesHandler}
 }
+
+// SetTransportsConfig assign transport config to Container
+func (tc *Container) SetTransportsConfig(clientConfig httphandlerConfig.Client) {
+	tc.TransportsConfig = clientConfig.Transports
+}
+
+// SelectTransport returns transport name by method, path and queryParams
+func (tc *Container) SelectTransport(method, path, queryParams string) (transportName string) {
+	_, transportName, ok := tc.TransportsConfig.GetMatchedTransport(method, path, queryParams)
+	if !ok {
```

Comment:
IMHO this branch is not necessary, as we can alwasy pick a transport for request (if none matches - pick the default)